usr/src/lib/libzfs/common/libzfs_dataset.c

```diff
@@ -1808,13 +1808,17 @@ zfs_prop_set_list(zfs_handle_t *zhp, nvlist_t *props)
 	ret = zfs_ioctl(hdl, ZFS_IOC_SET_PROP, &zc);
 
 	if (ret != 0) {
+		(void) zfs_standard_error(hdl, errno, errbuf);
```

Comment:
Why do we need to call `zfs_standard_error()` here, even if `zc_nvlist_dst_filled==TRUE`?  Maybe I'm reading the code wrong, but it seems like you'd get a repeated error message: one from this call, and then one from zfs_setprop_error().

Reply:
Oh.  You're right.  I didn't realise zfs_standard_error was actually printing stuff out.  I chose to do it this way because I thought it would handle the case where "errorprops" comes back empty, but I don't know if that's something that'll actually happen.  I'll have another go.